### PR TITLE
Add verification of config.yaml configuration to provide user-friendly prompts

### LINKED ISSUE #16404: Example improvement plan ||
Now the brand new example module has been initially completed, and the shardingshere-jdbc example is ready to use, but there are many areas that can be optimized.

Task:
- [x] Add verification of `config.yaml` configuration to provide user-friendly prompts
- [ ] Supports the combined generation of transaction, product, and operational modes

relate #12366

### PR DESCRIPTION
**Pull request by ShiFengCui**
Fixes #16404.

Changes proposed in this pull request:
- Add verification of config.yaml configuration to provide user-friendly prompts

### CODE REVIEW

File: examples/shardingsphere-example-generator/src/main/java/org/apache/shardingsphere/example/generator/core/ExampleGeneratorFactory.java [hunk 1]
```diff
@@ -53,13 +54,14 @@ private Configuration createTemplateConfiguration() throws IOException {
...
      * Generate directories and files by template.
-     * 
+     *
      * @throws TemplateException template exception
      * @throws IOException IO exception
      */
     @SuppressWarnings("unchecked")
     public void generate() throws TemplateException, IOException {
         YamlExampleConfiguration exampleConfiguration = swapConfigToObject();
+        new YamlExampleConfigurationValidator().validate(exampleConfiguration);
```
Comment (totalo) [inline examples/shardingsphere-example-generator/src/main/java/org/apache/shardingsphere/example/generator/core/ExampleGeneratorFactory.java:64]: Maybe the tool class is more suitable, we call the static method for verification

File: examples/shardingsphere-example-generator/src/main/java/org/apache/shardingsphere/example/generator/core/yaml/config/YamlExampleConfigurationValidator.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+public class YamlExampleConfigurationValidator {
+    
+    /**
+     * Verify the entrance.
+     *
+     * @param configuration configuration
+     */
+    public void validate(final YamlExampleConfiguration configuration) {
+        Map<String, ConfigurationData> valueMap = new HashMap<>();
+        valueMap.put("products", new ConfigurationData(configuration.getProducts(), Sets.newHashSet("jdbc", "proxy")));
```
Comment (totalo) [inline examples/shardingsphere-example-generator/src/main/java/org/apache/shardingsphere/example/generator/core/yaml/config/YamlExampleConfigurationValidator.java:44]: I think these can be abstracted into enumeration classes, or static properties

**Review by totalo — CHANGES_REQUESTED** [review]

**Comment by totalo** [comment]
hi @ShiFengCui ,Thanks for your contribution, here are some questions, please check.

**Comment by ShiFengCui** [comment]
> 嗨@ShiFengCui，感谢您的贡献，这里有一些问题，请检查。

OK, I got it. I'll adjust it

File: examples/shardingsphere-example-generator/src/main/java/org/apache/shardingsphere/example/generator/core/ExampleGeneratorFactory.java (hunk 1 above)
Comment (ShiFengCui) [reply examples/shardingsphere-example-generator/src/main/java/org/apache/shardingsphere/example/generator/core/ExampleGeneratorFactory.java:64]: ok

File: examples/shardingsphere-example-generator/src/main/java/org/apache/shardingsphere/example/generator/core/yaml/config/YamlExampleConfigurationValidator.java (hunk 2 above)
Comment (ShiFengCui) [reply examples/shardingsphere-example-generator/src/main/java/org/apache/shardingsphere/example/generator/core/yaml/config/YamlExampleConfigurationValidator.java:44]: ok

File: examples/shardingsphere-example-generator/src/main/java/org/apache/shardingsphere/example/generator/core/yaml/config/YamlExampleConfigurationValidator.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+import java.util.Set;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+
+/**
+ * Example configuration validator.
+ */
+public class YamlExampleConfigurationValidator {
```
Comment (totalo) [inline examples/shardingsphere-example-generator/src/main/java/org/apache/shardingsphere/example/generator/core/yaml/config/YamlExampleConfigurationValidator.java:32]: plz add final

**Review by totalo — CHANGES_REQUESTED** [review]

**Comment by totalo** [comment]
hi, @ShiFengCui ,Thanks for you nice job! There is one more small problem here, I will merge him after it is fixed.

**Review by totalo — APPROVED** [review]
LGTM.

**Pull request merged**